src/main/java/org/jenkinsci/plugins/docker/commons/KeyMaterialContext.java

```diff
@@ -0,0 +1,34 @@
+package org.jenkinsci.plugins.docker.commons;
+
+import hudson.FilePath;
+
+import javax.annotation.Nonnull;
+import java.io.Serializable;
+
+/**
+ * Represents the context within a {@link KeyMaterialFactory} can {@link KeyMaterialFactory#materialize()}
+ * {@link KeyMaterial} instances.
+ *
+ * @author Stephen Connolly
+ */
+public class KeyMaterialContext implements Serializable {
+    private static final long serialVersionUID = 1L;
+
+    @Nonnull
+    private final FilePath baseDir;
+
+    public KeyMaterialContext(@Nonnull FilePath baseDir) {
+        this.baseDir = baseDir;
+    }
+
+    /**
+     * Returns the base directory that can be used to {@link KeyMaterialFactory#materialize()}
+     * {@link KeyMaterial} instances.
+     *
+     * @return the base directory.
+     */
+    @Nonnull
+    public FilePath getBaseDir() {
+        return baseDir;
+    }
```

Comment:
@stephenc has it right ... these `KeyMaterial`/`KeyMaterialFactory` classes don't create that dir if I remember, so they should not be cleaning it up. They should only clan up dirs that they were responsible for creating. If something else created the `baseDir` then it is responsible for cleaning it up.